docs/docs/spec.md

```diff
@@ -42,33 +42,37 @@ The spec is not designed to enable fine-grained access restrictions (e.g. only r
 Icechunk requires that the storage system support the following operations:
 
 - **In-place write** - Strong read-after-write and list-after-write consistency is expected. Files are not moved or altered once they are written.
-- **Conditional write if-not-exists** - For the commit process to be safe and consistent, the storage system must guard against two files of the same name being created at the same time.
+- **Write-if-not-exists** - For creating new references.
+- **Conditional update** - For the commit process to be safe and consistent, the storage system must be able to atomically update a file only if the current version is known to the writer.
 - **Seekable reads** - Chunk file formats may require seek support (e.g. shards).
 - **Deletes** - Delete files that are no longer used (via a garbage-collection operation).
-- **Sorted List** - The storage system must allow the listing of directories / prefixes in a consistent sorted order.
+- **Sorted List** - The storage system must allow the listing of directories / prefixes in lexicographical order.
 
 These requirements are compatible with object stores, like S3, as well as with filesystems.
 
-The storage system is not required to support random-access writes. Once written, chunk and metadata files are immutable until they are deleted.
+The storage system is not required to support random-access writes. Once written, most files are immutable until they are deleted. The exceptions to this rule are: the repository configuration file, and branch refs.
```

Comment:
Also true for snapshots, no? For expiration...

Reply:
true! forgot about that one